File: VB/Dashboard_UserData/Form1.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports DevExpress.XtraEditors
Imports DevExpress.DashboardCommon
Imports System.Xml.Linq
Imports DevExpress.DashboardWin

Namespace Dashboard_UserData
	Partial Public Class Form1
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
			AddHandler dashboardViewer1.CustomizeDashboardTitle, AddressOf DashboardViewer1_CustomizeDashboardTitle
			dashboardViewer1.LoadDashboard("..\..\Data\Dashboard.xml")
		End Sub

		Private Sub DashboardViewer1_CustomizeDashboardTitle(ByVal sender As Object, ByVal e As CustomizeDashboardTitleEventArgs)
			Dim itemSave As DashboardToolbarItem = New DashboardToolbarItem(Sub(args)
				SaveDashboard()
			End Sub)
			itemSave.Caption = "Save Dashboard"
			e.Items.Add(itemSave)
			Dim itemLoad As DashboardToolbarItem = New DashboardToolbarItem(Sub(args)
				LoadDashboard()
			End Sub)
			itemLoad.Caption = "Load Dashboard"
			e.Items.Add(itemLoad)
		End Sub

		Private Sub SaveDashboard()
			Dim dashboard As New Dashboard()
			dashboard.LoadFromXml("..\..\Data\Dashboard.xml")

			Dim parameters As IList(Of DashboardParameterDescriptor) = dashboardViewer1.Parameters

			Dim xml As New XElement("ParametersInfo",
			    From parameter In parameters
			    Select New XElement("ParameterInfo", New XElement("Name", parameter.Name), New XElement("Value", parameter.SelectedValue), New XElement("Type", parameter.Type)))

			dashboard.UserData = xml
			dashboard.SaveToXml("..\..\Data\Dashboard.xml")
		End Sub

		Private Sub LoadDashboard()
			Dim dashboard As New Dashboard()
			dashboard.LoadFromXml("..\..\Data\Dashboard.xml")

			If dashboard.UserData IsNot Nothing Then
				Dim xml As XElement = dashboard.UserData

				Dim parsInfo As IList(Of XElement) = xml.Elements().ToList()
				Dim values As IList(Of Object) = New List(Of Object)()
				For Each parInfo As XElement In parsInfo
					If parInfo.Element("Type").Value = "DateTime" Then
						values.Add(Date.Parse(parInfo.Element("Value").Value))
					Else
						values.Add(parInfo.Element("Value").Value)
					End If
				Next parInfo

				Dim parameters1 As DashboardParameters = dashboardViewer1.Parameters

				dashboardViewer1.BeginUpdateParameters()
				For i As Integer = 0 To parameters1.Count - 1
					parameters1(i).SelectedValue = values(i)
				Next i
				dashboardViewer1.EndUpdateParameters()
			End If
		End Sub
	End Class
End Namespace

File: CS/Dashboard_UserData/Form1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using DevExpress.XtraEditors;
using DevExpress.DashboardCommon;
using System.Xml.Linq;
using DevExpress.DashboardWin;

namespace Dashboard_UserData {
    public partial class Form1 : XtraForm {
        public Form1() {
            InitializeComponent();
            dashboardViewer1.CustomizeDashboardTitle += DashboardViewer1_CustomizeDashboardTitle;
            dashboardViewer1.LoadDashboard(@"..\..\Data\Dashboard.xml");
        }

        private void DashboardViewer1_CustomizeDashboardTitle(object sender, CustomizeDashboardTitleEventArgs e)
        {
            DashboardToolbarItem itemSave = new DashboardToolbarItem((args) => {SaveDashboard();});
            itemSave.Caption = "Save Dashboard";
            e.Items.Add(itemSave);
            DashboardToolbarItem itemLoad = new DashboardToolbarItem((args) => { LoadDashboard(); });
            itemLoad.Caption = "Load Dashboard";
            e.Items.Add(itemLoad);
        }

        private void SaveDashboard()
        {
            Dashboard dashboard = new Dashboard();
            dashboard.LoadFromXml(@"..\..\Data\Dashboard.xml");

            IList<DashboardParameterDescriptor> parameters = dashboardViewer1.Parameters;

            XElement xml = new XElement("ParametersInfo",
                        from parameter in parameters
                        select new XElement("ParameterInfo",
                                    new XElement("Name", parameter.Name),
                                    new XElement("Value", parameter.SelectedValue),
                                    new XElement("Type", parameter.Type)));

            dashboard.UserData = xml;
            dashboard.SaveToXml(@"..\..\Data\Dashboard.xml");
        }

        private void LoadDashboard()
        {
            Dashboard dashboard = new Dashboard();
            dashboard.LoadFromXml(@"..\..\Data\Dashboard.xml");

            if (dashboard.UserData != null)
            {
                XElement xml = dashboard.UserData;

                IList<XElement> parsInfo = xml.Elements().ToList();
                IList<object> values = new List<object>();
                foreach (XElement parInfo in parsInfo)
                {
                    if (parInfo.Element("Type").Value == "DateTime")
                    {
                        values.Add(DateTime.Parse(parInfo.Element("Value").Value));
                    }
                    else
                    {
                        values.Add(parInfo.Element("Value").Value);
                    }
                }

                DashboardParameters parameters1 = dashboardViewer1.Parameters;

                dashboardViewer1.BeginUpdateParameters();
                for (int i = 0; i < parameters1.Count; i++)
                {
                    parameters1[i].SelectedValue = values[i];
                }
                dashboardViewer1.EndUpdateParameters();
            }
        }
    }
}
